File: frontend/src/types/monitoring.ts
export interface ScheduledRecrawl {
  id: string;
  project_id: string;
  start_url: string;
  crawl_settings_json: Record<string, unknown>;
  interval_hours: number;
  is_active: boolean;
  last_run_at: string | null;
  next_run_at: string;
  created_at: string;
}

export interface ScheduledRecrawlCreate {
  start_url: string;
  interval_hours: number;
  max_pages?: number;
  max_depth?: number;
  follow_subdomains?: boolean;
  include_pdfs?: boolean;
}

export interface Alert {
  id: string;
  project_id: string;
  project_name: string;
  crawl_snapshot_id: string;
  canonical_url: string;
  category: string;
  summary: string;
  acknowledged: boolean;
  created_at: string;
}

File: backend/app/schemas/monitoring.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ScheduledRecrawlCreate(BaseModel):
    start_url: str
    interval_hours: int = Field(default=24, ge=1, le=8760)
    max_pages: int | None = None
    max_depth: int | None = None
    follow_subdomains: bool = False
    include_pdfs: bool = False


class ScheduledRecrawlUpdate(BaseModel):
    is_active: bool | None = None
    interval_hours: int | None = Field(default=None, ge=1, le=8760)


class ScheduledRecrawlRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    start_url: str
    crawl_settings_json: dict
    interval_hours: int
    is_active: bool
    last_run_at: datetime | None
    next_run_at: datetime
    created_at: datetime


class AlertRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    crawl_snapshot_id: uuid.UUID
    canonical_url: str
    category: str
    summary: str
    acknowledged: bool
    created_at: datetime


class AlertWithProject(AlertRead):
    project_name: str
